pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -73,22 +77,44 @@ var supportedLimitResponseType = sets.NewString(
 // ValidateFlowSchema validates the content of flow-schema
 func ValidateFlowSchema(fs *flowcontrol.FlowSchema) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMeta(&fs.ObjectMeta, false, ValidateFlowSchemaName, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateFlowSchemaSpec(&fs.Spec, field.NewPath("spec"))...)
+	specPath := field.NewPath("spec")
+	switch fs.Name {
+	case flowcontrol.FlowSchemaNameExempt, flowcontrol.FlowSchemaNameCatchAll:
+		src := bootstrap.DefaultFlowSchemaExempt
+		if fs.Name == flowcontrol.FlowSchemaNameCatchAll {
+			src = bootstrap.DefaultFlowSchemaDefault
+		}
+		var rightSpec flowcontrol.FlowSchemaSpec
+		err := legacyscheme.Scheme.Converter().Convert(&src.Spec, &rightSpec, 0, &conversion.Meta{})
```

Comment:
we've been trying pruning the reference of legacyscheme for a period of time, and additionally validation should work upon internal types so we can't do conversion here

Reply:
When the installation code for our object types switches to using something other than `legacyscheme.Scheme` then this code can make the same switch.

Validation _does_ work on internal types.  That's why the conversion is here.  It is converting the external-type objects from `bootstrap` to the corresponding internal type so that a proper comparison can be made.  The unit tests are verifying that this is being done correctly.

Reply:
I refactored so that we have only one reference to `legacyscheme`, so the code here should get properly updated if/when our install code stops using `legacyscheme`.